src/System.Memory/src/System/SpanExtensions.cs

```diff
@@ -145,5 +145,47 @@ public static bool SequenceEqual(this ReadOnlySpan<byte> first, ReadOnlySpan<byt
             int length = first.Length;
             return length == second.Length && SpanHelpers.SequenceEqual(ref first.DangerousGetPinnableReference(), ref second.DangerousGetPinnableReference(), length);
         }
+
+        /// <summary>
+        /// Determines whether the specified sequence appears at the start of the span.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool StartsWith(this Span<byte> span, ReadOnlySpan<byte> value)
+        {
+            int valueLength = value.Length;
+            return valueLength <= span.Length && SpanHelpers.SequenceEqual(ref span.Slice(0, valueLength).DangerousGetPinnableReference(), ref value.DangerousGetPinnableReference(), valueLength);
```

Comment:
Yes, you are right.
The SequenceEqual method checks for length equality, the helper does not:
https://github.com/dotnet/corefx/blob/master/src/System.Memory/src/System/SpanExtensions.cs#L79